src/components/HbLoadingSpinner.tsx

```diff
@@ -0,0 +1,90 @@
+import random from "lodash/random";
+import React, { ReactNode, useContext, useMemo } from "react";
+import { Css } from "src/Css";
+import SpinnerGifBase64 from "./HbLoadingSpinner.base64";
+
+interface HbLoadingSpinnerProps {
+  /** Reverts loading text to `Loading...` if true. May override global noQuips by passing in noQuips={false}. */
+  noQuips?: boolean;
+  /** Contextually relevant quips, like "Where did I put that invoice??" on some invoice page. Will override global noQuips if non-empty. */
+  extraQuips?: ReadonlyArray<string>;
+  extraQuipsOnly?: boolean;
+  /** No text at all */
+  iconOnly?: boolean;
+}
+
+export function HbLoadingSpinner({ noQuips, extraQuips = [], extraQuipsOnly, iconOnly }: HbLoadingSpinnerProps) {
+  const ctx = useContext(HbLoadingSpinnerContext);
+
+  const quip = useMemo(() => {
+    // If quips are off globally but custom quips were provided, then override global
+    const forceQuips = extraQuips.length !== 0 || noQuips === false;
+    if ((ctx.noQuips && !forceQuips) || noQuips) {
+      return "Loading...";
+    }
+
+    const allQuips = extraQuipsOnly && extraQuips.length !== 0 ? extraQuips : [...ctx.quips, ...extraQuips];
+    return allQuips[random(allQuips.length - 1)];
+  }, [ctx.noQuips, ctx.quips, extraQuips, extraQuipsOnly, noQuips]);
+
+  return (
+    <div data-testid="hbLoadingSpinner" css={Css.df.fdc.jcc.aic.$}>
+      <img
+        src={SpinnerGifBase64}
+        data-chromatic="ignore" // Chromatic timing would snapshot this on different frames of the 24-frame gif, so ignore it
+        css={
+          Css.add({
+            // Image is white-on-black, so this flips it to black-on-white for our almost-always white backgrounds
+            filter: "invert(1)",
+          }).$
+        }
+        alt="loading"
+      />
+      {!iconOnly && (
+        <div
+          data-chromatic="ignore" // ignore spapshotting on randomized loading quips
+        >
```

Comment:
**question:** Thoughts on putting `data-testid`s on this or the image tag? Then we could have more explicit tests that check for a specific element, rather than just checking that any 1 or 2 elements are present. In my mind that'd be a more robust way of testing.